package.json

```diff
@@ -83,7 +86,7 @@
   "husky": {
     "hooks": {
       "pre-commit": "lint-staged",
-      "pre-push": "npm run lint && npm run test -- --bail && npm run test-dts -- --bail",
+      "pre-push": "npm run lint && npm run test -- --bail && npm run test-tsc -- --bail && npm run test-dts -- --bail",
```

Comment:
I think type checking needs fewer resources than unit testing. So we can run type checker before unit tests, in this way if there be a type error in tests, the push process will fail earlier without running extra tests.